src/compiler/types.ts

```diff
@@ -498,6 +498,7 @@ namespace ts {
         /* @internal */ original?: Node;                // The original node if this is an updated node.
         /* @internal */ startsOnNewLine?: boolean;      // Whether a synthesized node should start on a new line (used by transforms).
         /* @internal */ jsDocComments?: JSDoc[];        // JSDoc for the node, if it has any.
+        /* @internal */ jsDocCache?: (JSDoc | JSDocParameterTag)[]; // JSDoc for the node, plus JSDoc retrieved from its parents

```

Comment:
I improved the comment to explain that `jsDoc[Comments]` is the JSDoc syntactically on the node whereas `jsDocCache` is all applicable docs *and* `@param` tags.